### PR TITLE
Exclude test cases from code analysis

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
To avoid the false positives we intentionally introduce in the unit tests.

### CODE REVIEW

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM

File: azure-pipelines.yml [hunk 1]
```diff
@@ -22,6 +22,8 @@ variables:
     value: 'analyzers\tests\SonarAnalyzer.UnitTest\'
   - name: UnitTestResultsPath
     value: '$(UnitTestProjectPath)\TestResults'
+  - name: UnitTestExclusionsPattern
+    value: 'analyzers/tests/SonarAnalyzer.UnitTest/TestCases/**/*'
```
Comment (andrei-epure-sonarsource) [inline azure-pipelines.yml:26]: note that the path separator is different from `    value: 'analyzers\tests\SonarAnalyzer.UnitTest\'` in line 22

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]
I compared the END step logs from [a run on master](https://github.com/SonarSource/sonar-dotnet/runs/15736834535) vs [this run](https://github.com/SonarSource/sonar-dotnet/pull/7763) 

master
```
2023-08-09T06:12:06.4422381Z E11606CA-A186-4FEE-BA30-B1688747CD1A.sonar.tests=\
///..........many files.......
2023-08-09T06:12:06.4888310Z "C:\\ProgramData\\vsts-agent\\1\\s\\analyzers\\tests\\SonarAnalyzer.UnitTest\\TestCases\\AllBranchesShouldNotHaveSameImplementation.vb",\
2023-08-09T06:12:06.4888832Z "C:\\ProgramData\\vsts-agent\\1\\s\\analyzers\\tests\\SonarAnalyzer.UnitTest\\TestCases\\AlwaysSetDateTimeKind.vb",\
```


this PR
```
2023-08-09T09:50:35.7173878Z E11606CA-A186-4FEE-BA30-B1688747CD1A.sonar.tests=\
2023-08-09T09:50:35.7684627Z "C:\\ProgramData\\vsts-agent\\1\\s\\analyzers\\tests\\SonarAnalyzer.UnitTest\\TestCases\\AllBranchesShouldNotHaveSameImplementation.vb",\
2023-08-09T09:50:35.7685138Z "C:\\ProgramData\\vsts-agent\\1\\s\\analyzers\\tests\\SonarAnalyzer.UnitTest\\TestCases\\AlwaysSetDateTimeKind.vb",\
```

so it looks there hasn't been a change - maybe try with  `sonar.test.exclusions` ?

File: azure-pipelines.yml (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply azure-pipelines.yml:26]: I added it like this on purpose to match the documentation from: https://docs.sonarsource.com/sonarqube/latest/project-administration/analysis-scope/

**Comment by costin-zaharia-sonarsource** [comment]
From what I know the exclusions are processed only in the [analyzers](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs#L87-L90) and in the scanner cli.

Due to this, I'm not sure if this can be verified by looking at the setting you mentioned since the files are not filtered at that point. On the other hand, I did find this in the end step:
```
2023-08-09T09:50:47.1942136Z 09:50:47.193 DEBUG: 'analyzers\tests\SonarAnalyzer.UnitTest\TestCases\AlwaysSetDateTimeKind.vb' indexed as test with language 'vbnet'
```

I will change the configuration to `sonar.test.exclusions` and restart the job.

**Comment by costin-zaharia-sonarsource** [comment]
With the latest change, I cannot find anymore (in the logs) the test cases indexed.

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**